tcpclient/connection.go

```diff
@@ -27,6 +27,20 @@ const (
 	ConnectionError        ConnectionStatus = 4
 )
 
+// NewConnection initializes a connection given all values that are actually stored internally. This is just being used
```

Comment:
You could use it also in the connectionManagement...
Btw, this NewConnection is not really a NewConnection which would have a default status and metrics=0, I 'd says it's more a SetConnection 🤔

Reply:
Well, here we just adopt the standard naming convention: a constructor should follow the pattern New$ObjectName , and this is a very verbose constructor, not pretending to replace the standard instance creation (that could have the same name but less parameters, as yes, a new connection should have some defaults we are allowing the tests to override in this one)